### PR TITLE
Updated documentation and mbed-os version

### PR DESCRIPTION
**Pull request by SenRamakri**
1. README has been updated to capture more info on the example and reference.
2. mbed-os version updated to get the right version/repo.

### CODE REVIEW

**Comment by SenRamakri** [comment]
@kegilbert - Please review the README.

File: README.md [hunk 1]
```diff
@@ -29,6 +29,20 @@ cd mbed-os-example-crash-reporting
...
 ```
 
+#### Description
+
+This example demonstrates Mbed OS crash-reporting feature and the new APIs associated with it. If you look at the main.cpp, you will see a
+function with signature `void mbed_error_reboot_callback(mbed_error_ctx *error_context)`. This function is the callback called during Mbed-OS 
+initialization if the system rebooted due to a fatal error. This function is called with a pointer to `mbed_error_ctx` structure which captures the
+error context which triggered the reboot. When you run this program(see `main.cpp`), it will force a fatal error by triggering an exception,
+which causes an auto-reboot(warm-reset) of the system. Note that the system captures the error context in a special location in RAM before 
+triggering the auto-reboot. And during the reboot, the system will detect that the reboot was caused by a fatal error and it calls the
```
Comment (kegilbert) [inline README.md]:
```suggestion
triggering the auto-reboot. During the reboot, the system will detect that the reboot was caused by a fatal error and it calls the 
```

File: README.md [hunk 2]
```diff
@@ -29,6 +29,20 @@ cd mbed-os-example-crash-reporting
...
 
+#### Description
+
+This example demonstrates Mbed OS crash-reporting feature and the new APIs associated with it. If you look at the main.cpp, you will see a
+function with signature `void mbed_error_reboot_callback(mbed_error_ctx *error_context)`. This function is the callback called during Mbed-OS 
+initialization if the system rebooted due to a fatal error. This function is called with a pointer to `mbed_error_ctx` structure which captures the
+error context which triggered the reboot. When you run this program(see `main.cpp`), it will force a fatal error by triggering an exception,
+which causes an auto-reboot(warm-reset) of the system. Note that the system captures the error context in a special location in RAM before 
+triggering the auto-reboot. And during the reboot, the system will detect that the reboot was caused by a fatal error and it calls the 
+`mbed_error_reboot_callback()` with the pointer to error context captured in RAM. The callback function records the fact that a reboot
```
Comment (kegilbert) [inline README.md]:
```suggestion
`mbed_error_reboot_callback()` with a pointer to the error context captured in RAM. The callback function records the fact that a reboot
```

File: README.md [hunk 3]
```diff
@@ -29,6 +29,20 @@ cd mbed-os-example-crash-reporting
...
+occurred due to a fatal error(using a local variable named `reboot_error_happened`) and it captures the error context using `mbed_get_reboot_error_info()`
```
Comment (kegilbert) [inline README.md]: In general the 2-3 sentences here use the word error a lot it feels like. Not really a great away around that, just a thought as I was reading this.

**Review by kegilbert — COMMENTED** [review]
Left a minor comment, but looks good! Pinging @anotherbutler although sounds like she won't have a chance to get to it this week. She said it'd be fine to count her as an optional reviewer in this case.

File: README.md (hunk 3 above)
Comment (SenRamakri) [reply README.md]: @kegilbert - Thanks for the review, let me push a change to address these comments.

File: README.md [hunk 4]
```diff
@@ -29,6 +29,20 @@ cd mbed-os-example-crash-reporting
...
 
+#### Description
+
+This example demonstrates Mbed OS crash-reporting feature and the new APIs associated with it. If you look at the main.cpp, you will see a
+function with signature `void mbed_error_reboot_callback(mbed_error_ctx *error_context)`. This function is the callback called during Mbed-OS 
+initialization if the system rebooted due to a fatal error. This function is called with a pointer to `mbed_error_ctx` structure which captures the
+error-context which triggered the reboot. When you run this program(see `main.cpp`), it will force an exception,
+which causes an auto-reboot(warm-reset) of the system. Note that the system captures the error-context in a special location in RAM before 
+triggering the auto-reboot. During the reboot, the system will detect that the reboot was caused by a fatal error and it calls the 
+`mbed_error_reboot_callback()` with a pointer to error context captured in RAM. The callback function records the fact that a reboot
```
Comment (kegilbert) [inline README.md]:
```suggestion
`mbed_error_reboot_callback()` with a pointer to the error context captured in RAM. The callback function records the fact that a reboot
```

**Review by kegilbert — APPROVED** [review]
Last minor word addition but LGTM!

File: README.md (hunk 4 above)
Comment (SenRamakri) [reply README.md]: @kegilbert - I fixed this as well. I'll go ahead merge this now.
@AnotherButler - You may still comment on anything which doesn't look right, when you get a chance later. For now, Ill merge this as @kegilbert approved this.

**Pull request merged**